src/com/redhat/art/VersionTest.groovy

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/groovy
+//package com.redhat.art
+
+import com.redhat.art.Version
+
+class VersionTest extends GroovyTestCase {
+    def env
+    
+    def VersionTest(pipeline_env=null) {
+        env = pipeline_env
+    }
+
+    //protected Version v0
+    //protected Version v1
+    
+    //protected void setUp() {
+
+    //}
+    
+    void testConstructorString() {
+        def vs = new Version('3.2.1')
+
+        def expected = [3, 2, 1]
+        def actual = vs.v_array
+
+        assertEquals(actual, expected)
+    }
+
+    void testConstructorVersion() {
+        def vs0 = new Version('3.2.1')
+        def vs1 = new Version(vs0)
+
+        def expected = [3, 2, 1]
+        def actual = vs1.v_array
+
+        assertEquals(actual, expected)
+    }
+    
+    void testConstructorVersionStrin() {
+        def v_array = [3, 2, 1]
+        def vs1 = new Version(v_array)
+
+        def expected = [3, 2, 1]
+        def actual = vs1.v_array
+
+        assertEquals(actual, expected)
+    }
+
+    void testCompareTo() {
+
+        def inputs = [
+            [
+                testValue: new Version('3.2.1'),
+                greater: [
+                    new Version('3.2.1.1'),
```

Comment:
What's all this?

Reply:
All of the values in the list should test "greater" than 3.2.1.  3.2.1 is implicit 3.2.1.0  So 3.2.1.1 should test greater.